tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -189,9 +196,40 @@ private int Serialize(ref byte[] bytes, int offset, in SpanModel spanModel)
             offset += WriteTags(ref bytes, offset, in spanModel, tagProcessors);
             offset += WriteMetrics(ref bytes, offset, in spanModel, tagProcessors);
 
+            if (hasMetaStruct)
+            {
+                offset += WriteMetaStruct(ref bytes, offset, in spanModel);
+            }
+
+            return offset - originalOffset;
+        }
+
+        private int WriteMetaStruct(ref byte[] bytes, int offset, in SpanModel model)
+        {
+            int originalOffset = offset;
+            offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _metaStructBytes);
+
+            // We don't know the final count yet, depending on it, a different amount of bytes will be used for the header
+            // of the dictionary, so we need a temporal buffer
+
+            var temporalBytes = new byte[256];
+            var tagWriter = new TagWriter(this, null, temporalBytes, 0);
+            model.Span.Tags.EnumerateMetaStruct(ref tagWriter);
+            temporalBytes = tagWriter.Bytes;
+            Array.Resize(ref temporalBytes, tagWriter.Offset);
+
+            offset += MessagePackBinary.WriteMapHeader(ref bytes, offset, tagWriter.Count);
+            offset += MessagePackBinary.WriteRaw(ref bytes, offset, temporalBytes);
```

Comment:
"temporal" means something different than "temporary"

```suggestion
            // of the dictionary, so we need a temporary buffer

            var temporaryBytes = new byte[256];
            var tagWriter = new TagWriter(this, null, temporaryBytes, 0);
            model.Span.Tags.EnumerateMetaStruct(ref tagWriter);
            temporaryBytes = tagWriter.Bytes;
            Array.Resize(ref temporaryBytes, tagWriter.Offset);

            offset += MessagePackBinary.WriteMapHeader(ref bytes, offset, tagWriter.Count);
            offset += MessagePackBinary.WriteRaw(ref bytes, offset, temporaryBytes);
```